src/ds/bits.h

```diff
@@ -109,7 +109,9 @@ namespace snmalloc
 
     inline void pause()
     {
-#if defined(PLATFORM_IS_X86)
+#if defined(__clang__) && __has_builtin(__builtin_pause)
```

Comment:
Why do we need the `defined(__clang__)` check here?  If another compiler implements `__has_builtin` and `__builtin_pause` then we probably want to use that.  There's no short-circuit evaluation in the preprocessor, so this isn't guarding the `__has_builtin` check...